charger/ocpp/cp_core.go

```diff
@@ -111,6 +111,17 @@ func (cp *CP) StopTransaction(request *core.StopTransactionRequest) (*core.StopT
 
 	conn := cp.connectorByTransactionID(request.TransactionId)
 	if conn == nil {
+		if request.Timestamp != nil {
+			cp.log.ERROR.Printf("old StopTransaction received and rejecting it")
```

Comment:
```suggestion
```

Reply:
actually maybe just an INFO message should be produced - I mean I would really like to see in the log if a stop transaction has been missed - when this first happened to me I feared I had bricked it because evcc would be able to start anymore. I've changed it already locally and removed the Timestamp check but accidently the car loaded to 100% instead of 50 Friday to Saturday night, so I cannot test at the moment.
Once I have tested it I will update.

I also added a few lines to phasesFromChargeCurrents because the actual phases and the ones sent to it (and accepted) are often not matching, I would try to sort out the original phase switching before I commit it.

BTW:
I ran into a series of messages "Rejected" by the box on Thursday but I was able to fully identify yet what triggered it, I think the following had happened: the charger had just switched to status "A" because my wife had just pressed eject on the car at 15:08:09, but at 15:08:16 evcc still called enable with 0 after it had received the EVDisconnected:

sent JSON message to 30J-HZH-5R4: [2,"3080567554","SetChargingProfile",{"connectorId":1,"csChargingProfiles":{"chargingProfileId":1,"stackLevel":0,"chargingProfilePurpose":"TxProfile","chargingProfileKind":"Relative","chargingSchedule":{"chargingRateUnit":"A","chargingSchedulePeriod":[{"startPeriod":0,"limit":0}]}}}]

which resulted in

received JSON message from 30J-HZH-5R4: [3,"3080567554",{"status":"Rejected"}]

so maybe #12861 may need some addition checks, I've not implemented them yet, I fear that rolling that out may cause quite some problems and an optional reset parameters may be the safer option, I switched back to reset on Friday, because of the Rejects and the missed control we got home with 30k left. I'm also not sure if it does not have negative impact on phase switching, some comments in the code say that the loadpoint would be disabled when doing this, but thats not the case: ( see phases1p3p in occp.go)

Reply:
If you need details you can use TRACE.

Reply:
I think this is not a case for TRACE, because yes the messages are logged anyway, but this is the detection of an out of sync situation which in the logfile could only be found by doing multi line filtering to catch the startup state.

Reply:
Maybe, but normaler users will be helpless at this point. Please remove.

Reply:
I removed it the output. I also took out the check for Timestamp being nil, because it is not checked further and ends up in that if on a valid message anyway.